nodedb.go

```diff
@@ -87,22 +100,23 @@ func newNodeDB(db dbm.DB, cacheSize int, opts *Options, lg log.Logger) *nodeDB {
 	}
 
 	return &nodeDB{
-		logger:         lg,
-		db:             db,
-		batch:          db.NewBatch(),
-		opts:           *opts,
-		firstVersion:   0,
-		latestVersion:  0, // initially invalid
-		nodeCache:      cache.New(cacheSize),
-		fastNodeCache:  cache.New(fastNodeCacheSize),
-		versionReaders: make(map[int64]uint32, 8),
-		storageVersion: string(storeVersion),
+		logger:              lg,
+		db:                  db,
+		batch:               db.NewBatch(),
+		opts:                *opts,
+		firstVersion:        0,
+		latestVersion:       0, // initially invalid
+		legacyLatestVersion: 0,
+		nodeCache:           cache.New(cacheSize),
+		fastNodeCache:       cache.New(fastNodeCacheSize),
+		versionReaders:      make(map[int64]uint32, 8),
+		storageVersion:      string(storeVersion),
 	}
 }
 
 // GetNode gets a node from memory or disk. If it is an inner node, it does not
 // load its children.
-func (ndb *nodeDB) GetNode(nk *NodeKey) (*Node, error) {
+func (ndb *nodeDB) GetNode(nk []byte) (*Node, error) {
```

Comment:
This feels slightly odd to me. Shouldn't we have at least some type safety here?

Is the idea that theres two different formats of `[]byte` here? Would it be helpful to have a type alias here, or not really?

Reply:
we are using `NodeKey` type for the new format, I will add some descriptions, how about this?